### PR TITLE
fixed equals and hashcode for operators and value entities

### PR DESCRIPTION
**Pull request by rachit138**

### CODE REVIEW

File: hope-core/src/main/java/io/appform/hope/core/values/JsonPathValue.java [hunk 1]
```diff
@@ -52,4 +52,17 @@ public <T> T accept(Visitor<T> visitor) {
         return visitor.visit(this);
     }
 
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == null) { return false; }
```
Comment (santanusinha) [inline hope-core/src/main/java/io/appform/hope/core/values/JsonPathValue.java:57]: Follow standard formatting

File: hope-core/src/main/java/io/appform/hope/core/values/JsonPathValue.java [hunk 2]
```diff
@@ -52,4 +52,17 @@ public <T> T accept(Visitor<T> visitor) {
...
+        if (obj == this) { return true; }
+        if (obj.getClass() != getClass()) { return false; }
+        JsonPathValue other = (JsonPathValue) obj;
```
Comment (santanusinha) [inline hope-core/src/main/java/io/appform/hope/core/values/JsonPathValue.java:60]: `val`

File: hope-core/src/main/java/io/appform/hope/core/values/JsonPathValue.java [hunk 3]
```diff
@@ -52,4 +52,17 @@ public <T> T accept(Visitor<T> visitor) {
...
+        return Objects.equal(this.getPath(), other.getPath());
```
Comment (santanusinha) [inline hope-core/src/main/java/io/appform/hope/core/values/JsonPathValue.java:61]: Don't use guava. Use Objects.equals from java itself

File: hope-core/src/main/java/io/appform/hope/core/values/JsonPathValue.java [hunk 4]
```diff
@@ -52,4 +52,17 @@ public <T> T accept(Visitor<T> visitor) {
...
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hashCode(this.getPath());
```
Comment (santanusinha) [inline hope-core/src/main/java/io/appform/hope/core/values/JsonPathValue.java:66]: Use native `Objects.hashCode`

File: hope-core/src/test/java/io/appform/hope/core/operators/OperatorEqualityTest.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+import io.appform.hope.core.values.NumericValue;
+import io.appform.hope.core.values.StringValue;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class OperatorEqualityTest {
+
+    @Test
+    void OrEquality() {
```
Comment (santanusinha) [inline hope-core/src/test/java/io/appform/hope/core/operators/OperatorEqualityTest.java:14]: Use `val` for final variables everywhere

**Review by santanusinha — CHANGES_REQUESTED** [review]

File: hope-core/src/main/java/io/appform/hope/core/values/JsonPathValue.java (hunk 1 above)
Comment (rachit138) [reply hope-core/src/main/java/io/appform/hope/core/values/JsonPathValue.java]: formatter the file

File: hope-core/src/main/java/io/appform/hope/core/values/JsonPathValue.java (hunk 2 above)
Comment (rachit138) [reply hope-core/src/main/java/io/appform/hope/core/values/JsonPathValue.java]: done

File: hope-core/src/main/java/io/appform/hope/core/values/JsonPathValue.java (hunk 4 above)
Comment (rachit138) [reply hope-core/src/main/java/io/appform/hope/core/values/JsonPathValue.java:73]: done

**Review by santanusinha — APPROVED** [review]

**Pull request merged**